x-pack/elastic-agent/pkg/agent/operation/monitoring.go

```diff
@@ -232,6 +242,16 @@ func (o *Operator) getMonitoringFilebeatConfig(output interface{}) (map[string]i
 							},
 						},
 					},
+					{
+						"add_fields": map[string]interface{}{
+							"target": "datastream",
+							"fields": map[string]interface{}{
+								"type":      "logs",
+								"dataset":   "elastic.agent",
```

Comment:
This is wrong, should be `fmt.Sprintf("elastic.agent.%s", name)`

Reply:
@blakerouse updated